### DISCUSSION TITLE
Timeline

### DISCUSSION THREAD

**Discussion by JPhB** [peter88213/novelibre#36] [Ideas]
There is an older but excellent [Timeline software](https://thetimelineproj.sourceforge.net) written in Python. Unlike Aeon, it is free. Events can be dragged and contain text (but there is no editor).  I use it as a sort of Storyboard and was wondering how hard would it be to integrate it with a novelibre project.

![image](https://github.com/user-attachments/assets/cda6674a-117a-45d4-a2d4-a72a2ab10fea)

**Comment by peter88213** [comment]
You might want to take a look at the [nv_timeline](https://github.com/peter88213/nv_timeline/) plugin for novelibre. It can synchronize *.novx* and *.timeline* files, and It can also generate one format from the other and vice versa.

Basically, the *Timeline* canvas can be integrated with other applications written in Python. However, it uses a GUI framework that is not compatible with *novelibre*.

**Reply by JPhB** [reply]
I was looking at the Timeline viewer, then the Aeon plugin sync, and completely missed that one between them...

I created a project from my timeline and was hoping to see my Events showing up in novelibre. Synchronizing is probably needed? I did read the doc but don't see a Sync command.

**Reply by peter88213** [reply]
Everything you need to know is in the [online manual](https://peter88213.github.io/nvhelp-en/nv_timeline/) under "Conventions", but to make things easier, I have just added an extra note: To create a novelibre project with **File > New > Create from Timeline**, the events that are to become sections must have a “Section” label.

**Reply by peter88213** [reply]
The idea behind this is that only selected events should be included as sections in the novelibre project. There may also be background events, for example, that should not be synchronized. 

Here is a screenshot how to create the "Section" label with Timeline:

![Timeline screenshot](https://peter88213.github.io/nvhelp-en/_images/event01.png)

**Reply by peter88213** [reply]
Just as a side note: There is a similar tool for yWriter: https://peter88213.github.io/yw-timeline/

**Comment by JPhB** [comment]
Thanks for all this. But I'm not there yet. I have the Section label in some of my Events, yet no Section is created in Novelibre.

I did put a yw-timeline shortcut on my desktop, I drag my Timeline on it and it says a yw7 file is created. But when I look at it there is nothing. On the other hand, if I drag a yw7 project on the shortcut, a Timeline is created with all the Section as Events on top of each other.

**Reply by peter88213** [reply]
With yWriter, the label is not "Section", but "Scene", if I'm not wrong. Please consult the documentation. 

Be aware, this all appiles only if you create a new project from a timeline. When synchronizing an existent project, new sections/scenes only can be created with novelibre/yWriter. These then can be converted into Timeline events during synchronization. 

> a Timeline is created with all the Section as Events on top of each other.

Do your yWriter scenes have date/time information? If not, all events get the reference date, which may be the actual date/time. It's up to you to move them along the timeline. After syncing, the scene date/time then should be visible in yWriter. 

_nv_timeline_ is a bit more advanced; sections without date/time wil be converted into events ordered with a 1-day distance on the timeline, so it is easier to grab and drag them. This distance can be changed with a configuration file, see the manual.

**Reply by peter88213** [reply]
> it says a yw7 file is created. But when I look at it there is nothing. 

This sounds odd. Doesn't the message include the full path of the created file?

**Reply by JPhB** [reply]
You're right, it's Scene, and it works. I'll do more testing now. Thanks

**Comment by JPhB** [comment]
This is not a question directly related to Novelibre which I'd be happy to redirect through a more appropriate channel.

The font for the Timeline Description window is too small for me. Given that the Python source is available, I was wondering how hard would it be to change it, and recompile.

I contacted the developer but got no response.

**Reply by peter88213** [reply]
This is actually not the right place for feature requests concerning third-party software.
You can send a message to the "Timeline" mailing list:
thetimelineproj-user@lists.sourceforge.net

Anyway, it looks to me as if the Timeline project is not being actively developed at the moment. 

The sources are here: https://sourceforge.net/p/thetimelineproj/main/ci/default/tree/
Since "Timeline" is written in Python, there is no need to recompile the program. The text input box used is probably a standard widget of the "Wx" GUI framework, which should be able to be reconfigured in some way. However, I don't have any experience with Wx. 

If you cannot fix your problem yourself, I'm afraid you have to take the "Timeline" software as is. In case you generally experience problems with small fonts, you might want to change your screen resolution and/or use a larger screen.

**Reply by JPhB** [reply]
Thanks for responding.

As mentioned, I did send mail to the timelineproj mailing list, and got no response.

I did find a wx font file, timeline-2.9.0\source\timelinelib\wxgui\components\font.py, and in that file I changed the line "def __init__(self, point_size=12" to "def __init__(self, point_size=14".

I was then able to run Timeline using the  timeline-2.9.0\source\timeline.py file, but no change ensued in the GUI.

**Reply by peter88213** [reply]
So then you have managed to run Timeline from the sources. That's pretty good. 

`point_size` is a so-called "optional argument" of the Font constructor. What you changed is just the default value that is used if the caller doesn't deliver a point_size parameter. You will have look for the Font() instantiation and find out where the `point_size` parameter comes from. If you manage to change the source value, you will have a chance. But be aware, you don't know to which widgets this font size applies, and whether the enlarged text will still fit to all elements of the user interface.
